src/Device.js

```diff
@@ -15,6 +15,16 @@ import {
 
 import { Sensor, MeasurementInfo, SensorSpecs } from './Sensor.js';
 
+// Use JS implemented TextDecoder if it is not provided by the browser.
+let _TextDecoder;
+const encoding = require('text-encoding');
```

Comment:
I think this might break our esm module. I'll need to verify.

Reply:
oops- updated the package.json. thanks!

Reply:
yeah, tests are still failing. Sorry for the trouble... Please keep me posted!

Reply:
No problem, I'll pull this today and fix it up